src/models/WidgetTemplateImport.php

```diff
@@ -0,0 +1,212 @@
+use yii\base\InvalidArgumentException;
+use yii\base\InvalidConfigException;
+use yii\base\Model;
+use yii\helpers\FileHelper;
+use yii\helpers\Inflector;
+use yii\helpers\Json;
+
+/**
+ * --- PROPERTIES ---
+ *
+ * @property \yii\web\UploadedFile[] $tarFiles
+ *
+ * @author Elias Luhr
+ */
+class WidgetTemplateImport extends Model
+{
+    /**
+     * @var \yii\web\UploadedFile[]
+     */
+    public $tarFiles;
+
+    /**
+     * Import extract directory of the generated tar file
+     * If this is set with an alias, it will be automatically resolved later on
+     *
+     * @var string
+     */
+    public $importDirectory = '@runtime/tmp/dmstr/widgets/templates/import';
+
+    /**
+     * @return void
+     * @throws \yii\base\Exception if the export directory could not be created due to an php error
+     * @throws \yii\base\InvalidConfigException if either the import directory is not set or the export directory cannot
+     * be created due to permission errors
+     */
+    public function init()
+    {
+        parent::init();
+
+        // Ensure that the import directory for the generated files is set
+        if (empty($this->importDirectory)) {
+            throw new InvalidConfigException('$importDirectory must be set');
+        }
+
+        // Make sure that if an alias is set, it is resolved correctly
+        $this->importDirectory = \Yii::getAlias($this->importDirectory);
+
+        // recreate import directory
+        try {
+            FileHelper::removeDirectory($this->importDirectory);
+        } catch (ErrorException $e) {
+            \Yii::error($e->getMessage());
+            throw new InvalidConfigException("Error while recreating directory at: $this->importDirectory");
+        }
+        if (FileHelper::createDirectory($this->importDirectory) === false) {
+            throw new InvalidConfigException("Error while creating directory at: $this->importDirectory");
+        }
+    }
+
+    public function rules()
+    {
+        return [
+            ['tarFiles', 'file', 'skipOnEmpty' => false, 'extensions' => 'tar', 'maxFiles' => 20]
+        ];
+    }
+
+    public function upload(): bool
+    {
+        if ($this->validate()) {
+            foreach ($this->tarFiles as $file) {
+                if ($file->saveAs($this->importDirectory . DIRECTORY_SEPARATOR . $file->baseName . '.' . $file->extension) === false) {
+                    $this->addError('tarFiles', \Yii::t('widgets', 'Error while saving file'));
+                }
+            }
+            return true;
+        }
+
+        return false;
+    }
+
+    public function attributeLabels()
+    {
+        $attributeLabels = parent::attributeLabels();
+        $attributeLabels['tarFiles'] = \Yii::t('widgets', 'Files');
+        return $attributeLabels;
+    }
+
+    public function removeImportDirectory(): bool
+    {
+        try {
+            FileHelper::removeDirectory($this->importDirectory);
+            return true;
+        } catch (\ErrorException $e) {
```

Comment:
yii\base\ErrorException is imported. Is it intended that another ErrorException is used here?

Reply:
This line has been removed because it is no longer needed